aws/smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/apigateway/AuthorizerDefinition.java

```diff
@@ -187,6 +213,7 @@ public boolean equals(Object o) {
         AuthorizerDefinition that = (AuthorizerDefinition) o;
         return scheme.equals(that.scheme)
                && Objects.equals(type, that.type)
+               && authType.equals(that.authType)
```

Comment:
I think this could cause a NullPointerException. You should use `Objects.equals(authType, that.authType)`.

Reply:
Ah you're right, must have thought I was working with the `getAuthType` output